analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -309,6 +309,13 @@ public static Task<int> GetValueTask(int num)
             return Task.Run(() => num);
         }
 
+        public void TaskIsNull()
+        {
+            Task<long>[] arr;
+            var x = arr[0].Result;  // Error [CS0165]
```

Comment:
This doesn't reproduce the NRE on my machine with the original implementation 😢

Reply:
My bad. Fixed and verified that it fails. I was verifying for null previously :)